api/python/t4/api.py

```diff
@@ -266,26 +266,29 @@ def create_str(self, pkg_info):
                         f"{self._fmt_str(size_str, 15).rstrip(' ')}\t\n")
             return out
 
+    # get_package_registry may include '/.quilt', which Package.browse does not expect
     if registry is None or registry == 'local':
-        base_registry = get_package_registry(None)
+        package_registry = None
     else:
-        base_registry = get_package_registry(fix_url(registry))
+        package_registry = fix_url(registry)
 
-    named_packages = base_registry + '/named_packages'
+    base_registry = get_package_registry(package_registry)
+    registry = base_registry.rstrip('.quilt/')
```

Comment:
Please use a pathlib function instead. Also, strip and rstrip don't do what you think they do https://docs.python.org/2/library/string.html#string.rstrip

Reply:
You're right, using `rstrip` is dangerous. I always think it does what I think it does and always end up re-learned that it doesn't. 🙄 However, I used `rindex` instead, not a `pathlib` function. The problem `pathlib.Path([...]).parent` is that it doesn't behave well in the presence of a path that starts with `file://` or `s3://`; it replaces it with `file:/` or `s3:/` when you convert it back using e.g. `as_posix()`, which is wrong and causes downstream operations on the path, like `parse_s3_url`, to fail. So I have to stick to string operations here.